lockapi.pm

```diff
@@ -24,23 +24,30 @@ our @EXPORT = qw(mutex_create mutex_lock mutex_unlock mutex_try_lock mutex_wait
   barrier_create barrier_wait barrier_try_wait barrier_destroy);
 
 require bmwqemu;
-use mmapi qw(api_call get_job_info);
-use testapi 'record_info';
+use mmapi qw(api_call_2 get_job_info);
+use testapi ();
+
+use constant RETRY_COUNT    => $ENV{OS_AUTOINST_LOCKAPI_RETRY_COUNT}    // 7;
+use constant RETRY_INTERVAL => $ENV{OS_AUTOINST_LOCKAPI_RETRY_INTERVAL} // 10;
+use constant POLL_INTERVAL  => $ENV{OS_AUTOINST_LOCKAPI_POLL_INTERVAL}  // 5;
 
 sub _try_lock {
     my ($type, $name, $param) = @_;
-    # try up to 7 times
-    my $res = '';
-    for (1 .. 7) {
-        $res = api_call('post', "$type/$name", $param)->code;
-        return 1 if ($res == 200);
-        last unless $res == 410;
-        bmwqemu::fctinfo("Retry $_ of 7...");
-        sleep 10;
+
+    my $log_ctx               = "acquiring $type '$name'";
+    my %expected_return_codes = (200 => 1, 409 => 1, 410 => 1);
+    my $actual_return_code;
+    for (1 .. RETRY_COUNT) {
+        my $tx = api_call_2(post => "$type/$name", $param, \%expected_return_codes);
+        $actual_return_code = $tx->res->code;
+        last unless mmapi::handle_api_error($tx, $log_ctx, \%expected_return_codes);
+        last unless $actual_return_code == 410;
+        bmwqemu::fctinfo("Retry $_ of " . RETRY_COUNT);
+        sleep RETRY_INTERVAL;
```

Comment:
codecov reports this as uncovered. Please try to cover that or mark with `# uncoverable statement`